tests/PyPI-pytest/run_all.py

```diff
@@ -0,0 +1,381 @@
+        "url": "https://github.com/tartley/colorama.git",
+        "requirements_file": "requirements-dev.txt",
+        "ignored_tests": None,
+    },
+
+    "dateutil": {
+        "url": "https://github.com/dateutil/dateutil.git",
+        "requirements_file": "requirements-dev.txt",
+        "ignored_tests": None,
+    },
+
+    "idna": {
+        "url": "https://github.com/kjd/idna.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "jmespath": {
+        "url": "https://github.com/jmespath/jmespath.py.git",
+        "requirements_file": "requirements.txt",
+        "ignored_tests": None,
+    },
+
+    "markupsafe": {
+        "url": "https://github.com/pallets/markupsafe.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "numpy": {
+        "url": "https://github.com/numpy/numpy.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "pyasn1": {
+        "url": "https://github.com/etingof/pyasn1.git",
+        "requirements_file": "requirements.txt",
+        "ignored_tests": None,
+    },
+
+    "pytz": {
+        "url": "https://github.com/stub42/pytz.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "pyyaml": {
+        "url": "https://github.com/yaml/pyyaml.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "requests": {
+        "url": "https://github.com/kennethreitz/requests.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "rsa": {
+        "url": "https://github.com/sybrenstuvel/python-rsa.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "simplejson": {
+        "url": "https://github.com/simplejson/simplejson.git",
+        "requirements_file": None,
+        "ignored_tests": None,
+    },
+
+    "urllib3": {
+        "url": "https://github.com/urllib3/urllib3.git",
+        "requirements_file": "dev-requirements.txt",
+        "ignored_tests": (
+            "test/test_no_ssl.py",
+        )
+    },
+}
+
+
+
+def main():
+    cache_dir = os.path.join(getCacheDir(), "pypi-git-clones")
+
+    if not os.path.exists(cache_dir):
+        os.mkdir(cache_dir)
+
+    os.chdir(cache_dir)
+
+    search_mode = createSearchMode()
+
+    results = []
+
+    for package_name, details in sorted(packages.items()):
+        active = search_mode.consider(dirname=None, filename=package_name)
+
+        if not active:
+            continue
+
+        # skip these packages
+        if package_name in (
+            "numpy",
+            "pytz",
+            "pyyaml", # invalid command 'bdist_nuitka'
+        ):
+            if search_mode.abortIfExecuted():
+                break
+            continue
+
+        # clone package if not there, else update it
+        if not package_name in os.listdir(cache_dir):
+            os.system("git clone %s %s" % (details["url"], package_name))

```

Comment:
Never do `not x in y`, always use `x not in y`. more readable, faster, only better. I think PyLint may complain about it, not sure.

Reply:
got it